File: JEJE/impl/room_number_1457.c
// SPDX-License-Identifier: AGPL-3.0-or-later
// NOTICE: This code is restricted from being used for any AI or machine learning training purposes.
#include<stdio.h>

int arr[10];

int main() {
    
    int n;
    scanf("%d", &n);
    
    while (n > 0) {
        arr[n % 10]++;
        n /= 10;
    }

    arr[6] = (arr[6] + arr[9] + 1) / 2;
    arr[9] = 0;

    int ans = 0;
    for (int i = 0; i < 10; i++) {
        ans = (ans < arr[i] ? arr[i] : ans);
    }

    printf("%d", ans);
}

File: JEJE/impl/room_number_1457.py
# SPDX-License-Identifier: AGPL-3.0-or-later
# NOTICE: This code is restricted from being used for any AI or machine learning training purposes.
import sys

if __name__ == '__main__':
    n = int(sys.stdin.readline().rstrip())
    arr = [0] * 10

    while n > 0:
        arr[n % 10] += 1
        n //= 10
    
    arr[6] = (arr[6] + arr[9] + 1) // 2
    arr[9] = 0

    sys.stdout.write(str(max(arr)))